pyresample/geometry.py

```diff
@@ -547,6 +547,48 @@ def __init__(self, lons, lats, nprocs=1):
         elif lons.ndim > 2:
             raise ValueError('Only 1 and 2 dimensional swaths are allowed')
 
+    def copy(self):
+        """Copy the current swath."""
+        return SwathDefinition(self.lons, self.lats)
+
+    def aggregate(self, **dims):
+        """Aggregate the current swath definition by averaging.
+
+        For example, averaging over 2x2 windows:
+        `sd.aggregate(x=2, y=2)`
+        """
+        import pyproj
+        import dask.array as da
+
+        def do_transform(src, dst, lons, lats, alt):
+            import pyproj
+            x, y, z = pyproj.transform(src, dst, lons, lats, alt)
+            return np.dstack((x, y, z))
+
+        geocent = pyproj.Proj(proj='geocent')
+        latlong = pyproj.Proj(proj='latlong')
+        res = da.map_blocks(do_transform, latlong, geocent,
+                            self.lons.data, self.lats.data,
+                            da.zeros_like(self.lons.data), new_axis=[2],
+                            chunks=(self.lons.chunks[0], self.lons.chunks[1], 3))
+        res = DataArray(res, dims=['y', 'x', 'coord'], coords=self.lons.coords)
+        res = res.coarsen(**dims).mean()
```

Comment:
You said it doesn't need `coarsen` but it is using it right here. Right?

Reply:
This one does, not the AreaDefinition one